src/remoted/secure.c

```diff
@@ -242,6 +243,10 @@ void * rem_handler_main(__attribute__((unused)) void * args) {
     int wdb_sock = -1;
     mdebug1("Message handler thread started.");
 
+    // Reset all the agents' connection status in Wazuh DB
+    if (OS_SUCCESS != wdb_reset_agents_connection(&wdb_sock))
```

Comment:
Calling wdb_reset_agents_connection will generate a reset for each worker thread of Remoted. I suggest to move this ie to secure.c L150 (After all the module initializations)

Reply:
You are right. I moved the call just before the loop that creates all the threads close to the line of code that you suggested.